In [1]:
import math 
import string 
import sys

def get_words_from_line_list(text):  
    
    # Pre processing the given samples.
    
    pre_process = str.maketrans(string.punctuation+string.ascii_uppercase," "*len(string.punctuation)+string.ascii_lowercase)
    
    text = text.translate(pre_process) 
    
    # Splitting the text into individual words
    word_list = text.split() 

    b= []

    # Ignoring the words of length 2 as these do not make any contribution in preedicting the outcome.
    for i in word_list:
        if len(i) > 2:
            b.append(i)
            
    # checking if the words in our list are stopwords or not. If stopwords, we ignore them
    stopWords = ['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you', 'your', 'yours', 'yourself', 'yourselves', 'he', 'him', 'his', 'himself', 'she', 'her', 'hers', 'herself', 'it', 'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 'this', 'that', 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own', 'same', 'so', 'than', 'too', 'very', 's', 't', 'can', 'will', 'just', 'don', 'should', 'now']
    c = []
    for k in b:
        if k not in stopWords:
            c.append(k)
      
    # Using a Dictionary to find out the words and number of the times occuring
    D = {}   
    for j in c:    
        if j in D: 
            D[j] = D[j] + 1
              
        else: 
            D[j] = 1
    return D

f = ''' The easiest way to earn points with Fetch Rewards is to just shop for the items you already buy. If you have any eligible brands on your receipt, you will get points based on the total cost of the products. You do not need to cut out any coupons or scan individual UPCs. Just scan your receipt after you check out and we will find the savings for you. '''
f1 = ''' The easiest way to earn points with Fetch Rewards is to just shop for the products you already love. If you have any participating brands on your receipt, you'll get points based on the cost of the products. You don't need to clip any coupons or scan individual barcodes. Just scan each grocery receipt after you shop and we'll find the savings for you. '''
f2 = ''' We are always looking for opportunities for you to earn more points, which is why we also give you a selection of Special Offers. These Special Offers are opportunities to earn bonus points on top of the regular points you earn every time you purchase a participating brand. No need to pre-select these offers, we'll give you the points whether or not you knew about the offer. We just think it is easier that way. '''

s = get_words_from_line_list(f)
s1 = get_words_from_line_list(f1)
s2 = get_words_from_line_list(f2)

# Here we treat samples as vectors and calculate angle between two given document samples using cos(d) = s1.s2/(|s|*|s1|)

def Dotproduct(s,s1):
      
    sum = 0.0 
    for i in s:         
        if i in s1 : 
            sum += (s[i] * s1[i]) 
    return sum
                
n = Dotproduct(s,s1)
n1 = Dotproduct(s,s2)
d = math.sqrt(Dotproduct(s, s)*Dotproduct(s1, s1)) 
d1 = math.sqrt(Dotproduct(s, s)*Dotproduct(s2, s2)) 
distance = math.acos(n / d)
distance1 = math.acos(n1 / d1)

print("Similarity between Sample 1 and Sample 2 ",math.degrees(distance)) 
print("Similarity between Sample 1 and Sample 3 ",math.degrees(distance1))


Similarity between Sample 1 and Sample 2  35.22217634423552
Similarity between Sample 1 and Sample 3  74.65084498804694


d-range is between (0 and 90). When d is 0 degrees means the two documents are exactly identical and when d is 90 degrees means the two documents are different

Here as "Similarity between Sample 1 and Sample 2 is 35.22217634423552", that means they are near to identical
 and as "Similarity between Sample 1 and Sample 3 is 74.65084498804694", that means they are almost different. 
"""   Samples 1 and 2 are more similar than samples 1 and 3.  """